sklearn/base.py

```diff
@@ -449,6 +449,25 @@ def fit_transform(self, X, y=None, **fit_params):
             return self.fit(X, y, **fit_params).transform(X)
 
 
+class DensityMixin(object):
+    """Mixin class for all density estimators in scikit-learn."""
+    _estimator_type = "DensityEstimator"
+
+    def score(self, X, y = None):
+        """Returns the log likelihood under the model on the given data X.
+
+        Parameters
+        ----------
+        X : array-like, shape = (n_samples, n_features)
+
+        Returns
+        -------
+        score: float
+            sum of log probability of samples
+        """
+        return np.sum(self.score_samples(X))
```

Comment:
`score_samples` already does that (the user can always call sum by him/her self to get the likelihood of a batch).

I find `score_samples` not a very good name, maybe we should call rename the `score_samples` method `log_likelihood`.